### PR TITLE
Ccd 1025 data store

### PR DESCRIPTION
**Pull request by padmapriyanalam**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CCD-1025

### Change description ###
Fix for the Case Search functionality to address the Case Roles capability.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ccd/data/caseaccess/CachedCaseUserRepository.java [hunk 1]
```diff
@@ -49,4 +51,9 @@ public void revokeAccess(Long caseId, String userId, String caseRole) {
     public List<CaseUserEntity> findCaseUserRoles(List<Long> caseIds, List<String> userIds) {
         return caseUserRepository.findCaseUserRoles(caseIds, userIds);
     }
+
+    @Override
+    public Set<String> getCaseUserRolesByUserId(String userId) {
+        return caseUserRepository.getCaseUserRolesByUserId(userId);
```
Comment (smathangi) [inline src/main/java/uk/gov/hmcts/ccd/data/caseaccess/CachedCaseUserRepository.java:57]: Can you check during single search operation how many times it is hitting this method? We might need some request caching as well like others..

File: src/test/java/uk/gov/hmcts/ccd/domain/service/common/DefaultAuthorisedCaseDefinitionDataServiceTest.java [hunk 2]
```diff
@@ -68,6 +71,7 @@ void setUp() {
             when(caseTypeService.getCaseTypeForJurisdiction(CASE_TYPE, JURISDICTION)).thenReturn(caseTypeDefinition);
             when(caseTypeService.getCaseType(CASE_TYPE)).thenReturn(caseTypeDefinition);
             when(userRepository.getUserRoles()).thenReturn(userRoles);
+            when(caseUserRepository.getCaseUserRolesByUserId(anyString())).thenReturn(userRoles);
```
Comment (smathangi) [inline src/test/java/uk/gov/hmcts/ccd/domain/service/common/DefaultAuthorisedCaseDefinitionDataServiceTest.java:74]: can you match exact userId and return caseRoles object please

**Review by smathangi — COMMENTED** [review]
Please add an integration test at controller level for this scenario. (eg: UICaseSearchController#searchCases(..))

File: src/main/java/uk/gov/hmcts/ccd/data/caseaccess/CaseUserEntity.java [hunk 3]
```diff
@@ -20,7 +20,9 @@
         query = "SELECT cue FROM CaseUserEntity cue WHERE casePrimaryKey.caseDataId IN :case_data_ids "
             + "AND casePrimaryKey.userId IN :user_ids"),
     @NamedQuery(name = CaseUserEntity.GET_ALL_CASE_ROLES_BY_CASE_IDS,
-        query = "SELECT cue FROM CaseUserEntity cue WHERE casePrimaryKey.caseDataId IN :case_data_ids")
+        query = "SELECT cue FROM CaseUserEntity cue WHERE casePrimaryKey.caseDataId IN :case_data_ids"),
+    @NamedQuery(name = CaseUserEntity.GET_ALL_CASE_ROLES_BY_USER_ID,
+        query = "SELECT casePrimaryKey.caseRole FROM CaseUserEntity cue WHERE casePrimaryKey.userId = :userId")
```
[the remark this answers is not in the document]
Comment (padmapriyanalam) [reply src/main/java/uk/gov/hmcts/ccd/data/caseaccess/CaseUserEntity.java]: Added distinct clause

**Comment by padmapriyanalam** [comment]
> Please add an integration test at controller level for this scenario. (eg: UICaseSearchController#searchCases(..))

Added the Integration tests.

File: src/test/resources/mappings/security-case-type-1.json [hunk 4]
```diff
@@ -837,6 +837,23 @@
               "delete": false
             }
           ]
+        },
+        {
+          "id": "PENDING",
+          "name": "Done",
```
Comment (danlysiak) [inline src/test/resources/mappings/security-case-type-1.json:843]:
```suggestion
          "name": "Pending",
```

File: src/test/java/uk/gov/hmcts/ccd/endpoint/std/CaseSearchEndpointESSecurityIT.java [hunk 5]
```diff
@@ -451,8 +451,7 @@ void shouldOnlyReturnCasesSolicitorHasBeenGrantedAccessTo() throws Exception {
         CaseSearchResult caseSearchResult = executeRequest(searchRequest, CASE_TYPE_C, AUTOTEST1_SOLICITOR);
 
         assertAll(
-            () -> assertThat(caseSearchResult.getTotal(), is(1L)),
-            () -> assertThat(caseSearchResult.getCases().get(0).getReference(), is(1589460125872336L))
```
Comment (danlysiak) [inline src/test/java/uk/gov/hmcts/ccd/endpoint/std/CaseSearchEndpointESSecurityIT.java:455]: Is it possible to keep assertions verifying the exact cases please?

Comment (padmapriyanalam) [reply src/test/java/uk/gov/hmcts/ccd/endpoint/std/CaseSearchEndpointESSecurityIT.java:455]: Assertions has been changed.

**Review by danlysiak — APPROVED** [review]

**Review by smathangi — APPROVED** [review]

**Pull request merged**